nomad/structs/job.go

```diff
@@ -16,6 +16,46 @@ const (
 	JobServiceRegistrationsRPCMethod = "Job.GetServiceRegistrations"
 )
 
+type UIJob struct {
+	NamespacedID
+	Name          string
+	Type          string
+	NodePool      string
+	Datacenters   []string
+	Priority      int
+	Allocs        []JobStatusAlloc
+	SmartAlloc    map[string]int
```

Comment:
The whole "smart" notion in this PR is going to need some explanation in docstrings, etc. You've got an explanation buried in the implementation, but we should have it somewhere in the struct definitions.

Reply:
the "smart" wording was not intended to be the final product, and even stand out so much as to say "hey, do something about me! at least rename me, or something!"

I've been keeping it around in case @philrenaud could make use of it _instead of_ needing to include all the alloc stubs, or if it might be useful for some other as-yet-unknown purpose, but that has been less and less likely, so I think this extra little feature will probably just get stripped out.

Reply:
"smart" stuff removed in [f8ccbd0](https://github.com/hashicorp/nomad/pull/20130/commits/f8ccbd02da0141a2c3f3d18ae6b0635bd730178d)